src/main/java/com/garbagemule/MobArena/ArenaListener.java

```diff
@@ -511,6 +512,12 @@ else if (softRestore)
         }
     }
 
+    public void onProjectileLaunch(ProjectileLaunchEvent event) {
+        if (event.getEntity().getShooter() instanceof Player) {
+            arena.getProjectileManager().addProjectile(event.getEntity());
+        }
+    }
+
```

Comment:
This is where we need the sanity checks. Aside from being misplaced in the manager, there are some checks missing. Check out some of the other sanity checks in the other parts of the ArenaListener class for inspiration. Use short-circuiting (early returns) until you have a player/projectile combo that _this_ arena needs to track.

- Check if the arena is running (not 100% sure here, since lobby players could potentially throw tridents as well, in which case ignore this bit).
- Check if the shooter is a player.
- Check if the player is actually in _this_ arena (this part is pretty crucial). Either in the lobby or the arena. Don't track spectator projectiles.

If those checks pass, we have a projectile fired by a player who is in the arena, which means we need to add the projectile to the ProjectileManager.